### PR TITLE
First pass at FCM module

### PR DESCRIPTION
**Pull request by rogerhu**
Based on https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages

Still working on the last 4 tests... the Firebase Admin requires a service account with private key which makes this more complicated.

### CODE REVIEW

File: src/ParsePushAdapter.js [hunk 1]
```diff
@@ -32,9 +33,11 @@ export default class ParsePushAdapter {
           this.senderMap[pushType] = new APNS(pushConfig[pushType]);
           break;
         case 'android':
-        case 'fcm':
           this.senderMap[pushType] = new GCM(pushConfig[pushType]);
           break;
+        case 'fcm':
+          this.senderMap[pushType] = new FCM(pushConfig[pushType]);
```
Comment (rogerhu) [inline src/ParsePushAdapter.js]: going to take this out for now... getting this code checked in and verified before it gets used is probably safer than making it the default

**Comment by rogerhu** [comment]
I'm not a Jasmine expert...can someone look at the exceptions and tell me how to mock out the Firebase send calls properly? :)

**Comment by flovilmart** [comment]
@rogerhu, if I recall correctly, node-GCM works with firebase no?

**Comment by rogerhu** [comment]
More context here:

https://firebase.googleblog.com/2017/11/whats-new-with-fcm-customizing-messages.html

**Pull request closed without merging**